pkg/operator/apis/monitoring/v1/types.go

```diff
@@ -1374,13 +1337,52 @@ type RulesStatus struct {
 	// TODO: add status information.
 }
 
+// CommonScrapeEndpointSpec
+type endpointParams struct {
+	jobName          string
+	scheme           string
+	path             string
+	interval         string
+	timeout          string
+	params           map[string][]string
+	metricRelabeling []RelabelingRule
+	relabelCfgs      []*relabel.Config
+	httpCfg          config.HTTPClientConfig
+	discoveryCfgs    discovery.Configs
+	limits           *ScrapeLimits
+}
+
+// ScrapeNodeEndpoint specifies a Prometheus metrics endpoint on a node to scrape.
```

Comment:
Could you summarize here, the difference between ScrapeEndpoint and this (e.g TLS fields are not present)?  (for users & future devs)